custom_components/solcast_solar/__init__.py

```diff
@@ -187,7 +187,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     solcast.headers = {
         'Accept': 'application/json',
-        'User-Agent': 'ha-solcast-solar-integration/'+version.replace('v','')[:version.rfind('.')-1]
+        'User-Agent': 'ha-solcast-solar-integration/'+version[:version.rfind('.')-1]
```

Comment:
```
>>> version='4.2.4'
>>> version[:version.rfind('.')-1]
'4.'
>>> version[:version.rfind('.')]
'4.2'
>>>
```

Reply:
```
>>> version='4.2.4'
>>> version[:version.rfind('.')-1]
'4.'
>>> version[:version.rfind('.')]
'4.2'
>>> version.rfind('.')
3
>>> version.rfind('.')-1
2
>>>
```

Reply:
WTH? 🤦🏻

Reply:
arr[:stop]              # items from the beginning through stop-1

Reply:
I am an idiot. Failed to copy the .py to my rig before restart...

Reply:
So the -1 is unneeded.

Reply:
No I did not. I failed to copy it to the _correct_ rig.

I have too many Home Assistants.